reframe/frontend/cli.py

```diff
@@ -97,8 +97,13 @@ def dep_lines(u, *, prefix, depth=0, lines=None, printed=None,
 
             location = inspect.getfile(type(u.check))
             if detailed:
-                details = (f' [variant: {u.check.variant_num}, '
-                           f'file: {location!r}]')
+                details_fields = [
+                    f'variant: {u.check.variant_num}',
+                    f'file: {location!r}'
+                ]
+                if hasattr(u.check, 'descr') and u.check.descr:
+                    details_fields.append(f'description: {u.check.descr}')
```

Comment:
I would always add the description even if it's not set. I would do the following:

1. If the attribute is not present (pretty uncommon, as it would only happen if a user specifically does `descr = required` in a test), I would print `<undefined>`
2. If it's empty, I would print `<n/a>`.

Also as a style comment, we always leave a blank line after if/for/while etc. blocks.